### ISSUE TITLE
[1.4] UPnP runs even if not enabled/selected in Serverconfig.txt

### ISSUE THREAD

**Issue by Solxanich** [tModLoader/tModLoader#1749]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

Latest public-1.4-alpha

### OS

Windows

### Platform

Steam

### Description

Original: 'Starting tModLoader from steam, then starting the server from start-tModLoader.bat was the issue. Starting the server first then startingting TML using start-tModLoader.bat was the fix.'

'I restarted, started the server, then joined with the client and it was gravy. After stopping the server and starting again I was unable to connect from the client. After restarting again, I was able to join the server once more, but stopping then restarting the server exhibits the same behavior. If you look at the server log I attached there is a bit about a gateway exception and this is shown in the terminal as well.'

Restarted referring to the computer.

### Log File

![image](https://user-images.githubusercontent.com/59670736/129623243-0c4cf41b-803d-4379-ba1d-b5970b2be50c.png)
[server (1).log](https://github.com/tModLoader/tModLoader/files/6995103/server.1.log)

'For the record UPnP is off in the server config and in my security gateway'

### Expected Behavior

Able to host and connect to server via public IP

### Actual Behavior

Unable to host and connect to server via public IP

**Comment by DarioDaF** [comment]
Not sure if this should be a separate issue but:
```
[17:01:51] [4/WARN] [tML]: Silently Caught Exception: 
System.Net.Sockets.SocketException: Cercando di utilizzare un argomento del puntatore in una chiamata, il sistema ha rilevato un indirizzo del puntatore non valido.
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
   at System.Net.Sockets.Socket.SendTo(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags, EndPoint remoteEP)
   at System.Net.Sockets.Socket.SendTo(Byte[] buffer, EndPoint remoteEP)
   at UwUPnP.Gateway.TryNew(IPAddress ip, Gateway& gateway) in tModLoader\Terraria\UwUPnP\Gateway.cs:line 66
   at UwUPnP.UPnP.StartListener(IPAddress ip) in tModLoader\Terraria\UwUPnP\UPnP.cs:line 77
   at UwUPnP.UPnP.<>c__DisplayClass15_0.<FindGateway>b__1() in tModLoader\Terraria\UwUPnP\UPnP.cs:line 69
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   at System.Threading.Tasks.Task.ExecuteEntryUnsafe(Thread threadPoolThread)
   at System.Threading.Tasks.Task.ExecuteFromThreadPool(Thread threadPoolThread)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```
This is an "intended error" of `UwUPnP.UPnP` while checking for a gateway on point to point cards with associated ips (in my case PtP VirtualBox network adapter).

```
[17:01:54] [9/WARN] [tML]: Silently Caught Exception: 
System.Net.Sockets.SocketException: Impossibile stabilire la connessione. Risposta non corretta della parte connessa dopo l'intervallo di tempo oppure mancata risposta dall'host collegato.
   at System.Net.Sockets.Socket.Receive(Byte[] buffer)
   at UwUPnP.Gateway.TryNew(IPAddress ip, Gateway& gateway) in tModLoader\Terraria\UwUPnP\Gateway.cs:line 68
   at UwUPnP.UPnP.StartListener(IPAddress ip) in tModLoader\Terraria\UwUPnP\UPnP.cs:line 77
   at UwUPnP.UPnP.<>c__DisplayClass15_0.<FindGateway>b__1() in tModLoader\Terraria\UwUPnP\UPnP.cs:line 69
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   at System.Threading.Tasks.Task.ExecuteEntryUnsafe(Thread threadPoolThread)
   at System.Threading.Tasks.Task.ExecuteFromThreadPool(Thread threadPoolThread)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```
This is an "intended error" of `UwUPnP.UPnP` while checking for a gateway on all non internet connected cards with associated ips (WILL always happen for each non internet card).

```
[17:03:41] [Main Thread/WARN] [tML]: Silently Caught Exception: 
System.Net.WebException: The remote server returned an error: (500) Internal Server Error.
   at System.Net.HttpWebRequest.SendRequest(Boolean async)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at System.Net.HttpWebRequest.SendRequest(Boolean async)
   at System.Net.HttpWebRequest.GetResponse()
   at UwUPnP.Gateway.GetResponse(HttpWebRequest request) in tModLoader\Terraria\UwUPnP\Gateway.cs:line 212
   at UwUPnP.Gateway.RunCommand(String action, ValueTuple`2[] args) in tModLoader\Terraria\UwUPnP\Gateway.cs:line 174
   at UwUPnP.Gateway.GetGenericPortMappingEntry(Int32 portMappingIndex) in tModLoader\Terraria\UwUPnP\Gateway.cs:line 262
   at UwUPnP.UPnP.GetGenericPortMappingEntry(Int32 portMappingIndex) in tModLoader\Terraria\UwUPnP\UPnP.cs:line 60
   at NATUPNPLib.UPnPNAT.MappingCollection.GetEnumerator()+MoveNext() in tModLoader\Terraria\NATUPNPLib\UPnPNAT.cs:line 49
   at Terraria.Netplay.OpenPort(Int32 port) in tModLoader\Terraria\Netplay.cs:line 151
   at Terraria.Netplay.InitializeServer() in tModLoader\Terraria\Netplay.cs:line 280
   at Terraria.Netplay.StartServer() in tModLoader\Terraria\Netplay.cs:line 239
   at Terraria.Main.DedServ_PostModLoad(Boolean& reloadMods) in tModLoader\Terraria\Main.cs:line 4507
   at Terraria.Main.DedServ() in tModLoader\Terraria\Main.cs:line 4056
   at Terraria.Program.LaunchGame_(Boolean isServer) in tModLoader\Terraria\Program.cs:line 207
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs) in tModLoader\Terraria\Program.cs:line 171
   at MonoLaunch.Main(String[] args) in tModLoader\Terraria\MonoLaunch.cs:line 52
```
Seems reasonable to happen anytime a Gateway is actually found (so only one iff the UPnP works) during enumeration of mapping options (`NATUPNPLib.UPnPNAT.MappingCollection.GetEnumerator` only `yeald break` exit path so pretty much this error IS NEEDED for this implementation to work).

```
[17:03:47] [Main Thread/WARN] [tML]: Silently Caught Exception: 
System.Collections.Generic.KeyNotFoundException: The given key 'NewInternalPort' was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at NATUPNPLib.UPnPNAT.MappingCollection.GetEnumerator()+MoveNext() in tModLoader\Terraria\NATUPNPLib\UPnPNAT.cs:line 50
   at Terraria.Netplay.OpenPort(Int32 port) in tModLoader\Terraria\Netplay.cs:line 151
   at Terraria.Netplay.InitializeServer() in tModLoader\Terraria\Netplay.cs:line 280
   at Terraria.Netplay.StartServer() in tModLoader\Terraria\Netplay.cs:line 239
   at Terraria.Main.DedServ_PostModLoad(Boolean& reloadMods) in tModLoader\Terraria\Main.cs:line 4507
   at Terraria.Main.DedServ() in tModLoader\Terraria\Main.cs:line 4056
   at Terraria.Program.LaunchGame_(Boolean isServer) in tModLoader\Terraria\Program.cs:line 207
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs) in tModLoader\Terraria\Program.cs:line 171
   at MonoLaunch.Main(String[] args) in tModLoader\Terraria\MonoLaunch.cs:line 52
```
Exact same as above because it will propagate from an empty dictionary responce while requesting the first "out of bounds" mapping (see `GetResponse` and `ret` being left empty by the exception right above this one).

On shutdown some similar error might appear due to reiterating an enumerator, but fixing the above would remove also this problem.

**Comment by Solxanich** [comment]
Addressed in #1771
